backend/src/main/java/ca/bc/gov/hlth/hnweb/controller/EligibilityController.java

```diff
@@ -54,23 +53,31 @@
 
 	private static final Logger logger = LoggerFactory.getLogger(EligibilityController.class);
 
+	@Deprecated
 	private static final String DATE_FORMAT_yyyyMMdd = "yyyyMMdd";
 
 	@Autowired
 	private EligibilityService eligibilityService;
 	
-	@GetMapping("/check-eligibility")
-	public ResponseEntity<CheckEligibilityResponse> checkEligibility(@RequestParam(name = "phn", required = true) String phn,
-			@DateTimeFormat(iso = ISO.DATE) @RequestParam(name = "eligibilityDate", required = false) Date eligibilityDate) {
-		logger.info("checkEligibility request - phn: {} date: {}", phn, eligibilityDate);
+	@Autowired
+	private MSHDefaults mshDefaults;
+	
+	@PostMapping("/check-eligibility")
+	public ResponseEntity<CheckEligibilityResponse> checkEligibility(@Valid @RequestBody CheckEligibilityRequest checkEligibilityRequest) {
 
 		try {
-			CheckEligibilityResponse checkEligibilityResponse = eligibilityService.checkEligibility(phn, eligibilityDate);			
-			ResponseEntity<CheckEligibilityResponse> response = ResponseEntity.ok(checkEligibilityResponse);		
+			R15Converter converter = new R15Converter(mshDefaults);
+			R15 r15 = converter.convertRequest(checkEligibilityRequest);
+			Message message = eligibilityService.checkEligibility(r15);
+			
+			CheckEligibilityResponse checkEligibilityResponse = converter.convertResponse(message);
+			
+			ResponseEntity<CheckEligibilityResponse> response = ResponseEntity.ok(checkEligibilityResponse);
+
 			logger.info("checkEligibility response: {} ", checkEligibilityResponse);
 			return response;	
 		} catch (Exception e) {
-			throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Bad /checkEligibility request", e);
+			throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Bad /check-eligibility request", e);
```

Comment:
I imagine we'll add some more specific error handling once we're actually connecting to downstream systems? Right now I think the only time an error could happen is with a bad request?

Reply:
Yes. This was just added as an example of the currently recommended way to handle errors in Spring.